diff --git a/node_modules/@once-ui-system/core/dist/components/Checkbox.js b/node_modules/@once-ui-system/core/dist/components/Checkbox.js
index d77f766..594eccb 100644
--- a/node_modules/@once-ui-system/core/dist/components/Checkbox.js
+++ b/node_modules/@once-ui-system/core/dist/components/Checkbox.js
@@ -1,13 +1,12 @@
 "use client";
 import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
-import { useState, useEffect, forwardRef } from "react";
+import { useState, useEffect, useId, forwardRef } from "react";
 import classNames from "classnames";
 import { Flex, Icon, InteractiveDetails } from ".";
 import styles from "./SharedInteractiveStyles.module.scss";
-const generateId = () => `checkbox-${Math.random().toString(36).substring(2, 9)}`;
 const Checkbox = forwardRef(({ style, className, isChecked: controlledIsChecked, isIndeterminate = false, onToggle, disabled, ...props }, ref) => {
     const [isChecked, setIsChecked] = useState(controlledIsChecked || false);
-    const [checkboxId] = useState(generateId());
+    const checkboxId = `checkbox-${useId()}`;
     useEffect(() => {
         if (controlledIsChecked !== undefined) {
             setIsChecked(controlledIsChecked);
